woocommerce/woocommerce-ios
# [TTPUK] Use country code for About Tap to Pay screen

Closes: #10890
Merge after: #10910 and #10900
N.B. Because I merged trunk and a previous branch, this'll look massive until the above are merged. It's not really, but the review should probably wait until those are merged.



## Description


We'll add Tap to Pay on iPhone support for UK WooPayments merchants. This PR uses the CountryCode enum value from the CardPresentPaymentsConfiguration instead of a raw string, for safer country checks and simpler localization in the app layer.

## Testing instructions



Using a build from Xcode, and a UK-based WooPayments store, on an iPhone XS or newer running iOS 16+:

1. Navigate to Menu > Payments
2. Observe that `About Tap to Pay on iPhone` is visible – tap it
3. Try the `Learn more about card readers` button – observe that it opens an authenticated web view to the correct card reader page for the UK.
4. Close the web view and tap `Learn more` at the bottom (related to Tap to Pay) – observe that it opens a webview to the documentation for Tap to Pay
5. Tap the `Set up Tap to Pay on iPhone` button and run through the set up flow, checking that it works.


Repeat the above test with a US WooPayments store. On step 3, observe that there is no button and the `Important information` section is not shown.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/be954f2e-06fd-492e-a813-0acce7b31541



---
